neo/rawio/neuronexusrawio.py

```diff
@@ -0,0 +1,281 @@
+        stream_id = '0'  # hard-coded see note above
+        for channel_index, channel_name in enumerate(channel_info["chan_name"]):
+            channel_id = channel_info["ntv_chan_name"][channel_index]
+            # 'ai0' indicates analog data which is stored as microvolts
+            if channel_info["chan_type"][channel_index] == "ai0":
+                units = "uV"
+            # 'd' means digital. Per discussion with neuroconv users the use of
+            # 'a.u.' makes the units clearer
+            elif channel_info["chan_type"][channel_index][0] == "d":
+                units = "a.u."
+            # aux channel
+            else:
+                units = "V"
+
+            signal_channels.append(
+                (
+                    channel_name,
+                    channel_id,
+                    self._sampling_frequency,
+                    BINARY_DTYPE,
+                    units,
+                    1,  # no gain
+                    0,  # no offset
+                    stream_id,
+                )
+            )
+
+        signal_channels = np.array(signal_channels, dtype=_signal_channel_dtype)
+
+        stream_ids = np.unique(signal_channels["stream_id"])
+        signal_streams = np.zeros(stream_ids.size, dtype=_signal_stream_dtype)
+        signal_streams["id"] = [str(stream_id) for stream_id in stream_ids]
+        for stream_index, stream_id in enumerate(stream_ids):
+            name = stream_id_to_stream_name.get(int(stream_id), "")
+            signal_streams["name"][stream_index] = name
+
+        # No events
+        event_channels = []
+        event_channels = np.array(event_channels, dtype=_event_channel_dtype)
+
+        # No spikes
+        spike_channels = []
+        spike_channels = np.array(spike_channels, dtype=_spike_channel_dtype)
+
+        # Put all the necessary info in the header
+        self.header = {}
+        self.header["nb_block"] = 1
+        self.header["nb_segment"] = [1]
+        self.header["signal_streams"] = signal_streams
+        self.header["signal_channels"] = signal_channels
+        self.header["spike_channels"] = spike_channels
+        self.header["event_channels"] = event_channels
+
+        # Add the minimum annotations
+        self._generate_minimal_annotations()
+
+    def _get_signal_size(self, block_index, seg_index, stream_index):
+
+        # All streams have the same size so just return the raw_data size
+        return self._raw_data.size
+
+    def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop, stream_index, channel_indexes):
+
+        if i_start is None:
+            i_start = 0
+        if i_stop is None:
+            i_stop = self._get_signal_size(block_index, seg_index, stream_index)
+
+        if channel_indexes is None:
+            channel_indexes = slice(None)
+
+        raw_data = self._raw_data
+        return raw_data[i_start:i_stop, channel_indexes]
```

Comment:
If I remember correctly, double slicing sometimes do not play so well when different kind of slicing

```suggestion
        raw_data = self._raw_data[i_start:i_stop, :]
        if channel_indexes is None:
            raw_data = raw_data[:, channel_indexes]
        return raw_data
```

Reply:
Works for me. I didn't know double-slicing could be bad (I think we do it in other places), so happy to split it!

Reply:
It is a bit fuzzy but some stuff like mixing slice and bool mask in fancy indexing can be strange. Maybe it is not anymore the case. But maybe we have this in many places in neo. SO no worry

Reply:
I'm already fixing it :) We were missing one thing.